pennylane/qnode_new/qnode.py

```diff
@@ -0,0 +1,935 @@
+            self.output_conversion = np.asarray
+            self.output_dim = len(res)
+            res = tuple(res)
+        else:
+            raise QuantumFunctionError("A quantum function must return either a single measured observable "
+                                       "or a nonempty sequence of measured observables.")
+
+        # check that all returned observables have a return_type specified
+        for x in res:
+            if x.return_type is None:
+                raise QuantumFunctionError(
+                    "Observable '{}' does not have the measurement type specified.".format(x))
+
+        # check that all ev's are returned, in the correct order
+        if res != tuple(self.obs_queue):
+            raise QuantumFunctionError(
+                "All measured observables must be returned in the order they are measured.")
+
+        # check that no wires are measured more than once
+        m_wires = list(w for ob in res for w in ob.wires)
+        if len(m_wires) != len(set(m_wires)):
+            raise QuantumFunctionError('Each wire in the quantum circuit can only be measured once.')
+
+        self.ops = self.queue + list(res)
+        del self.queue
+        del self.obs_queue
+
+        # True if op is a CV, False if it is a discrete variable (Identity could be either)
+        are_cvs = [isinstance(op, plo.CV) for op in self.ops if not isinstance(op, plops.Identity)]
+        if not all(are_cvs) and any(are_cvs):
+            raise QuantumFunctionError(
+                "Continuous and discrete operations are not allowed in the same quantum circuit.")
+
+        # TODO: we should enforce plugins using the Device.capabilities dictionary to specify
+        # whether they are qubit or CV devices, and remove this logic here.
+        #self.type = 'cv' if all(are_cvs) else 'qubit'  #: str: circuit type, in {'cv', 'qubit'}
+
+
+    def _sort_args(self, args, kwargs):
+        """NEW: Sort the quantum function arguments to positional and keyword-only.
+
+        When the QNode is evaluated via :meth:`~QNode.__call__`, ``kwargs`` may also contain
+        differentiable POSITIONAL_OR_KEYWORD arguments given using the keyword syntax,
+        in addition to KEYWORD_ONLY arguments.
+
+        This method moves all positional arguments from ``kwargs`` to ``args``,
+        and substitutes missing positional arguments with special singleton markers.
+
+        Args:
+            args (tuple[Any]): positionally given arguments
+            kwargs (dict[str, Any]): keyword-given arguments
+
+        Returns:
+            tuple[Any], dict[str, Any]: positional arguments, auxiliary arguments
+        """
+        n_args = len(args)  # number of positionally-given args
+        pos = list(args)
+        for name, s in self.func.sig.items():
+            # is it positional?
+            if s.par.kind <= inspect.Parameter.POSITIONAL_OR_KEYWORD:
```

Comment:
What does <= mean in this context? Is `POSITIONAL_OR_KEYWORD` an IntEnum?

Reply:
Yes.